core/domain/skill_domain.py

```diff
@@ -653,3 +653,48 @@ def to_dict(self):
             'skill_model_created_on': self.skill_model_created_on,
             'skill_model_last_updated': self.skill_model_last_updated
         }
+
+
+class SkillsMastery(object):
+    """Domain object for an Oppia Skills Mastery."""
+
+    @classmethod
+    def get_model_id(cls, user_id, skill_id):
+        """Returns model id corresponding to user and skill.
+
+        Args:
+            user_id: str. The user id of the logged in user.
+            skill_id: str. The unique id of the skill.
+
+        Returns:
+            str. The model id corresponding to user and skill.
+        """
+        return user_id + '-' + skill_id
+
+    def __init__(self, user_id, skill_id, degree_of_mastery):
+        """Constructs a SkillMastery domain object.
+
+        Args:
+            user_id: str. The user id of the logged in user.
+            skill_id: str. The id of the skill.
+            degree_of_mastery: float. The progress of user for the
+                corresponding skill.
+        """
+
+        self.id = self.get_model_id(user_id, skill_id)
```

Comment:
I don't think you need the ID at all in the domain layer. The ID is just used by the storage layer to construct the model key and store the data at a particular location, but it does not contribute to any of the logic operations in the domain layer of the codebase.

Reply:
The ID here is model_id that corresponds to the user and skill. We'll need this to specify the particular model key.

Reply:
Yes, I am saying that the generation logic for the model key should happen in the storage layer and not the domain layer.

Take a look at the other models; there are instances where we construct the model id from multiple components. In those model classes there is usually a method that takes the individual components and constructs the composite ID. It should be private to the model class, i.e. an internal detail of it.

Reply:
Done